src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs

```diff
@@ -0,0 +1,247 @@
+﻿using AutoMapper;
+using Spd.Resource.Repository;
+using Spd.Resource.Repository.Document;
+using Spd.Resource.Repository.Incident;
+using Spd.Resource.Repository.Licence;
+using Spd.Resource.Repository.LicenceApplication;
+using Spd.Resource.Repository.ServiceTypes;
+using Spd.Resource.Repository.WorkerLicenceCategory;
+using Spd.Utilities.FileStorage;
+using Spd.Utilities.Printing.BCMailPlus;
+using Spd.Utilities.Shared.Exceptions;
+using Spd.Utilities.Shared.Tools;
+using System.Drawing;
+using System.Drawing.Drawing2D;
+using System.Drawing.Imaging;
+using System.Net;
+using System.Text.Json.Serialization;
+
+namespace Spd.Manager.Printing.Documents.TransformationStrategies;
+
+internal class PersonalLicencePreviewTransformStrategy(ILicenceApplicationRepository licAppRepository,
+    ILicenceRepository licRepository,
+    IServiceTypeRepository serviceTypeRepository,
+    IDocumentRepository documentRepository,
+    IMainFileStorageService fileStorageService,
+    IWorkerLicenceCategoryRepository workerLicenceCategoryRepository,
+    IIncidentRepository incidentRepository,
+    IMapper mapper)
+    : BcMailPlusTransformStrategyBase<PersonalLicencePreviewTransformRequest, LicencePreviewJson>(Jobs.SecurityWorkerLicense)
+{
+    protected override async Task<LicencePreviewJson> CreateDocument(PersonalLicencePreviewTransformRequest request, CancellationToken cancellationToken)
+    {
+        //following is for personal licence.
+        LicenceListResp lics = await licRepository.QueryAsync(new LicenceQry() { LicenceId = request.LicenceId }, cancellationToken);
+        if (lics == null || !lics.Items.Any())
+        {
+            throw new ApiException(HttpStatusCode.BadRequest, "no licence found for the licenceId");
+        }
+        LicenceResp lic = lics.Items.First();
+        LicencePreviewJson preview = mapper.Map<LicencePreviewJson>(lic);
+        var serviceTypeListResp = await serviceTypeRepository.QueryAsync(
+                new ServiceTypeQry(null, Enum.Parse<ServiceTypeEnum>(preview.LicenceType)), cancellationToken);
+        string name = serviceTypeListResp.Items.First().ServiceTypeName;
+        preview.LicenceType = name;

```

Comment:
Please assign this value directly in previous line

Reply:
ok, fixed